lib/parser/deserializer.js

```diff
@@ -414,8 +415,8 @@ function deserializeObject(buffer, index, options, isArray) {
         buffer[index + stringSize - 1] !== 0
       )
         throw new Error('bad string length in bson');
-      // symbol is deprecated - upgrade to string.
-      object[name] = buffer.toString('utf8', index, index + stringSize - 1);
+      const symbol = buffer.toString('utf8', index, index + stringSize - 1);
+      object[name] = promoteValues ? symbol : new BSONSymbol(symbol);
```

Comment:
nit: cant we benefit from short circuiting here and not have to do toString on the buffer if we might just toss it?

Reply:
hm, how would we short circuit here? At this point we are definitely going to use the string, it's just a matter of whether we return just the string or wrap it in a `BSONSymbol` instance.

Reply:
I'm blind 🤦‍♂ nvm LGTM